plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+    if (hContext == NULL) {
+        flb_error("Failed to create renderContext");
+        goto cleanup;
+    }
+
+    if (EvtRender(hContext,
+                  handle,
+                  EvtRenderEventValues,
+                  BUFSIZE,
+                  wBuffer,
+                  &dwBufferUsed,
+                  &count) != FALSE){
+        status = ERROR_SUCCESS;
+    }
+    else {
+        status = GetLastError();
+    }
+
+    if (status != ERROR_SUCCESS) {
+        flb_error("failed to query RenderContextValues");
+        goto cleanup;
+    }
+    pValues = (PEVT_VARIANT)wBuffer;
+
+    hMetadata = EvtOpenPublisherMetadata(
+            NULL, // TODO: Remote handle
+            pValues[0].StringVal,
+            NULL,
+            MAKELCID(langID, SORT_DEFAULT),
+            0);
+    if (hMetadata == NULL) {
+        goto cleanup;
+    }
+
+    wMessage = get_message(hMetadata, handle, message_size);
+
+#undef BUFSIZE
+
+cleanup:
+    if (hContext) {
+        EvtClose(hContext);
+    }
+
+    if (hMetadata) {
+        EvtClose(hMetadata);
+    }
+
+    return wMessage;
+}
+
+int get_string_inserts(EVT_HANDLE handle, PEVT_VARIANT *values, UINT *prop_count, unsigned int *string_inserts_size)
+{
+    PEVT_VARIANT pValues;
+    DWORD dwBufferSize = 0;
+    DWORD dwBufferSizeUsed = 0;
+    DWORD dwPropCount = 0;
+    BOOL succeeded = FLB_TRUE;
+
+    EVT_HANDLE hContext = EvtCreateRenderContext(0, NULL, EvtRenderContextUser);
+    if (hContext == NULL) {
+        flb_error("Failed to create renderContext");
+        succeeded = FLB_FALSE;
+        goto cleanup;
+    }
+
+    // Get the size of the buffer
+    EvtRender(hContext, handle, EvtRenderEventValues, 0, NULL, &dwBufferSize, &dwPropCount);
+    pValues = (PEVT_VARIANT)flb_malloc(dwBufferSize);
+
+    succeeded = EvtRender(hContext,
+                          handle,
+                          EvtRenderContextValues,
+                          dwBufferSize,
+                          pValues,
+                          &dwBufferSizeUsed,
+                          &dwPropCount);
+
+    if (!succeeded) {
+        flb_error("Failed to get string inserts with %d\n", GetLastError());
+        goto cleanup;
+    }
+
+    *prop_count = dwPropCount;
+    *values = pValues;
+    *string_inserts_size = dwBufferSize;
+
+cleanup:
+
+    if (hContext != NULL)
+        EvtClose(hContext);
+
+    return succeeded;
+}
+
+static int winevtlog_next(struct winevtlog_channel *ch, int hit_threshold)
+{
+    EVT_HANDLE hEvents[SUBSCRIBE_ARRAY_SIZE];
+    DWORD count = 0;
+    DWORD status = ERROR_SUCCESS;
+
+    /* If subscription handle is NULL, it should return false. */
+    if (!ch->subscription) {
+        flb_error("Invalid subscription is passed");
+        return FLB_FALSE;
+    }
+
+    if (hit_threshold) {
+        return FLB_FALSE;
+    }
+
+    if (!EvtNext(ch->subscription, SUBSCRIBE_ARRAY_SIZE, hEvents, INFINITE, 0, &count)) {
```

Comment:
Could you move this so the result is saved to a variable which is used in the conditional?

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/a02c29553e0fffc9eeb847eecd10a21f3767f9e6